### DISCUSSION TITLE
Query for products with search params returns all products.

### DISCUSSION THREAD

**Discussion by adm-bjmn** [wp-graphql/wp-graphql#3174] [WooCommerce]
Im trying filter a query with a string as follows : 

```
query SearchProducts {
  products(where: {search: "testing"}) {
    nodes {
      id
      name
    }
    found
  }
}

```
However this always returns all products. 
```

  "data": {
    "products": {
      "nodes": [
        {
          "id": "cHJvZHVjdDo3Mw==",
          "name": "A different product"
        },
        {
          "id": "cHJvZHVjdDo3Mg==",
          "name": "testing"
        },
        {
          "id": "cHJvZHVjdDo0OA==",
          "name": "tesing product"
        }
      ],
      "found": 3
    }
  },

```
Ive tried other search strings to target words only found in descriptions etc, Ive also tried querying with the REST API and adding in a search param and I get back the expected results so I have ruled out a woo commerce configuration as it seems the tables are indexed appropriately for the REST API to get the correct products based on the search.

All other 'where' option seem to work also: eg maxPrice, attribute etc. return expected results per query.

Only the search by string seems to be not providing the expected results. 

This behaviour happens in the IDE on the wordpress back end as well as when making queries from my Nextjs front end.

Does anyone have a suggestion as to where I can look to solve this. 
I don't want all the data per product so avoiding the rest API would be great but if this doesn't work then i guess it doesn't work.

**Comment by justlevine** [comment]
Sounds like you're using [wp-graphql-woocommerce](https://github.com/wp-graphql/wp-graphql-woocommerce), so I suggest you start by searching there.

**Reply by adm-bjmn** [reply]
Okay thank you
